doc/source/data/glossary.rst

```diff
@@ -0,0 +1,154 @@
+
+        .. doctest::
+
+            >>> import ray
+            >>> dataset = ray.data.range(10)
+            >>> dataset.random_shuffle().take_all()  # doctest: +SKIP
+            [4, 7, 1, 5, 2, 3, 8, 6, 9, 0]
+
+        If :meth:`~ray.data.Dataset.random_shuffle()` is slow, perform a local shuffle.
+        It’s lower quality but more performant. To perform a local shuffle, call methods
+        like :meth:`~ray.data.Dataset.iter_batches` and specify ``local_shuffle_buffer_size``.
+
+        .. doctest::
+
+            >>> import ray
+            >>> dataset = ray.data.range(10)
+            >>> next(iter(dataset.iter_batches(local_shuffle_buffer_size=5)))  # doctest: +SKIP
+            [2, 6, 7, 8, 4, 9, 3, 1, 5, 0]
+
+        To learn more about shuffling data, read :ref:`Shuffling Data <air-shuffle>`.
+
+    Datasets (library)
+        A library for distributed data processing.
+
+        Datasets isn’t intended as a replacement for more general data processing systems.
+        Its utility is as the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray.
+
+        To learn more about Ray Datasets, read :ref:`Key Concepts <dataset_concept>`.
+
+    Dataset (object)
+        A class that represents a distributed collection of data.
+
+        :class:`Dataset <ray.data.Dataset>` exposes methods to read, transform, and consume data at scale.
+
+        To learn more about Datasets and the operations they support, read the :ref:`Datasets API Reference <data-api>`.
+
+    Datasource
+        A :class:`~ray.data.Datasource` specifies how to read and write from
+        a variety of external storage and data formats.
+
+        Examples of Datasources include :class:`~ray.data.datasource.ParquetDatasource`,
+        :class:`~ray.data.datasource.ImageDatasource`,
+        :class:`~ray.data.datasource.TFRecordDatasource`,
+        :class:`~ray.data.datasource.CSVDatasource`, and
+        :class:`~ray.data.datasource.MongoDatasource`.
+
+        To learn more about Datasources, read :ref:`Creating a Custom Datasource <custom_datasources>`.
+
+    Record
+        A single data item.
+
+        If your dataset is :term:`tabular <Tabular Dataset>`, then records are :class:`TableRows <ray.data.row.TableRow>`.
+        If your dataset is :term:`simple <Simple Dataset>`, then records are arbitrary Python objects.
+        If your dataset is :term:`tensor <Tensor Dataset>`, then records are `NumPy ndarrays <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`_.
+
+    Schema
+        The data type of a dataset.
+
+        If your dataset is :term:`tabular <Tabular Dataset>`, then the schema describes
+        the column names and data types. If your dataset is :term:`simple <Simple Dataset>`,
+        then the schema describes the Python object type. If your dataset is :term:`tensor <Tensor Dataset>`, then the schema isn’t relevant.
```

Comment:
Hmm the schema is still somewhat useful for "tensor datasets", since they give you the per-element tensor shape and the underlying data type. I know that the schema leaks the internal column name, but the column type is still somewhat useful.